fish-pepper/run-java-sh/fp-files/run-java.sh

```diff
@@ -127,9 +127,9 @@ calc() {
       return x > y ? x : y
     }
     function round(x) {
-      return int(x + 0.5)
+      return x + 0.5
     }
-    {printf "%d\n",'"${formula}"'}
+    {print '"int(${formula})"'}
```

Comment:
@astefanutti actually look at this diff to the previous version there was always an _int_ at the end (also for log2 and ceil) because of the given %d. I don't mind to read-add int() to round and ceil, but this is not necessary, or ?

Reply:
It's just for internal evaluation of the formula prior to the `int` cast, where it will slightly vary the result if you call `ceil` or `round`.

Reply:
@astefanutti sorry, I didn't get this: When you e.g call `$(calc 'round($1*$2/100)' "100" "100"` then in the original version this was translated to 

```
echo "100 100" | awk '
    function ceil(x) {
      return x % 1 ? x + 1 : x
    }
    function log2(x) {
      return log(x)/log(2)
    }
    function max2(x, y) {
      return x > y ? x : y
    }
    function round(x) {
      return x + 0.5
    }
    {printf "%d\n", '"round($1*$2/100))"'}
'
```

so at the end the result was stuffed in a `%d` which simply truncates any float. How is this different to 

```
     { print '"int(round($1*$2/100)"' }
```

?? For ceil I agree, that the internal `int` might be required but for round it isn't or ( `int(int(x + 0.5))` is the same as `int(x + 0.5)`)

Reply:
@rhuss, like:
```
int(round(0.2 + 0.5) + round(0.2 + 0.5)) = int(0.7 + 0.7) = int(1.4) = 1
int(round(0.2 + 0.5) + round(0.2 + 0.5)) = int(0 + 0) = 0
```

Reply:
@astefanutti ok, got it (if we consider combined cases, which we currently dont have). I already re-added int() back to round, so we should be save now.